exe/wallet/Main.hs

```diff
@@ -311,15 +314,19 @@ exec execServer manager args
         :: forall a. ()
         => (a -> BL.ByteString)
         -> ClientM a
+        -> Bool
         -> IO ()
-    runClient encode cmd = do
+    runClient encode cmd silentMode = do
         port <- args `parseArg` longOption "port"
         let env = mkClientEnv manager (BaseUrl Http "localhost" port "")
         res <- runClientM cmd env
         case res of
             Right a -> do
-                TIO.hPutStrLn stderr "Ok."
-                BL8.putStrLn (encode a)
+                if silentMode then
+                    TIO.hPutStr stderr ""
+                else do
+                    TIO.hPutStrLn stderr "Ok."
+                    BL8.putStrLn (encode a)
```

Comment:
What's the rational / requirement for the `silentMode` :thinking: ?
Also, avoid `Bool` where you can and favor a concrete data-type that conveys the semantic better:

```hs
data Mode = Silent | Verbose
```

Although here, I don't think we need this ?

Reply:
Instead, I'll suggest to:

- split `runClient` into two smaller functions:
    - `sendRequest`
    - `handleResponse`

This way, you may re-use `sendRequest` easily to check that a wallet exists without doing all the standard handling.